src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureCredentialsProvider.java

```diff
@@ -144,13 +150,23 @@ private static Collection<IdCredentials> fetchCredentials() {
                     }
                     case "sshUserPrivateKey": {
                         String usernameSecretTag = tags.get("username-is-secret");
+                        String passphraseID = tags.get("passphrase-id");
+                        Secret passphrase = null;
                         boolean usernameSecret = false;
                         if (StringUtils.isNotBlank(usernameSecretTag)) {
                             usernameSecret = Boolean.parseBoolean(usernameSecretTag);
                         }
+                        if (StringUtils.isNotBlank(passphraseID)) {
+                            try {
+                                passphrase = new KeyVaultSecretRetriever(client, keyVaultURL + "secrets/" + passphraseID).get();
+                            } catch (Exception e) {
+                                LOG.log(Level.WARNING, "Could not find passphrase " + passphraseID + " in KeyVault.  Defaulting to null");
+                                passphrase = null;
+                            }
```

Comment:
could we log a warning and skip the whole secret instead? not sure if we want to create a broken secret?